src/particle.cpp

```diff
@@ -353,6 +358,40 @@ void Particle::event_death()
   }
 }
 
+void Particle::energy_delivert_in_cell()
+{
+  // Update pht_storage array
+  pht_storage()[coord(0).cell] += E_last() - E();
```

Comment:
These new functions appear to be very short, and you could redesign the code so that you only need to call this once. You call it right before the return in the various branches of `sample_photon_reaction` branches. Rather than calling it at the end of each branch, can't you instead just call it after calling `sample_photon_reaction`?

Or, rather, I think since this would in principle be the same for each particle type, we'd probably want to call this in the `event_collide` function right around the `score_collision_tally` function. The problem here is that we would be assuming we only lose energy at collisions, and if we ever do charged particle transport correctly, that would be the wrong assumption. But maybe that bridge could be crossed at a later date with a cautionary TODO statement.

Also, the comment "update the pht_storage array" is not helpful at all because anyone with a basic knowledge of C++ can immediately tell that's what this line of code is doing. A more helpful comment would be something along the lines of "tally particle energy decrement to top-level universe cell". One thing I notice here is that this will probably not work with the distribcell approach (which would require taking `cell_instance()` into account in your `pht_storage` array).

Reply:
Several comments here:
- I like the idea of calling it only once after `sample_photon_reactions` a lot. Thanks for that idea. 
I placed this function as you suggested in `Particle::event_collide()` in line 257 directly after the `score_collision_tally`. Since we don't want the electrons to account into this value, I added another if condition:
```cpp
if (type() == ParticleType::photon){energy_delivered_in_cell();}
```
- I also added a TODO. I think we should "cross this bridge later".
- when you say these functions appear to be very short, do you think we still should work here with functions or simply call
```cpp
pht_storage()[coord(0).cell] += E_last() - E();
```
?
- I changed the comment to 
``` // Adds the energy the particle loses during the collision. ```
but we can also use yours. 
- It currently will not work with the distribcell approach, which could be interesting for users. My thought was that in a first version this tally only works with the `CellFilter` and `EnergyFilter`. What do you think?